feat(content): Flesh out "Fixing Common Errors" with more examples (#264)
railwayapp/docs: src/docs/troubleshoot/fixing-common-errors.md

## Application Error: This application failed to respond

> https requets will fail with status code 503 Bad Gateway

Reviewer: There's a typo
```suggestion
The HTTPS request will fail with status code 503 Bad Gateway
```

> The HTTPS request will fail with status code 503 Bad Gateway

Reviewer 2: Super nitpick: I'd add this below the screenshot so that it reads better

**This approach is not recommended**.
</Banner>

> ## Some examples for a few frameworks / servers:

Reviewer: The elements of this section should all share the same heading (`###`) to have a consistent result
So, the following will need to be amended accordingly:
* Node / Express
* Node / Nest
* Python / Gunicorn

Author: this was intentional

Reviewer 2: Also curious about this, are you going for "most common language/framework" as the header?

I think having them consistent with h3/`###` makes sense

#### Node / Express

```javascript
// Use PORT provided in environment or default to 3000
const port = process.env.PORT || 3000;

// Listen on `port` and 0.0.0.0
app.listen(port, "0.0.0.0", function () {
  // ...
});
```

#### Node / Nest

```javascript
// Use PORT provided in environment or default to 3000
const port = process.env.PORT || 3000;

// Listen on `port` and 0.0.0.0
async function bootstrap() {
  // ...
  await app.listen(port, "0.0.0.0");
}
```

### Node / Next

> next needs an edditional flag to listen on `port`

Reviewer: ```suggestion
next needs an additional flag to listen on `port`
```

```bash
next start --port ${PORT-3000}
```

#### Python / Gunicorn

`gunicorn` listens on `0.0.0.0` and the `PORT` environment variable by default.
There is no additional configuration necessary.

```bash
gunicorn main:app
```

### Python / Uvicorn

`uvicorn` needs additional configuration flags to listen on `0.0.0.0` and `PORT`

```bash
uvicorn main:app --host 0.0.0.0 --port $PORT
```

### Go / net/http

This example is for the stdlib net/http but similar methods can be used with other frameworks

```golang
func main() {
  // ...

  // Use PORT provided in environment or default to 3000
  var port = envPortOr("3000")

  log.Fatal(http.ListenAndServe(port, handler))
}

> // returns environment PORT or port parameter, both will be prfixed with ":"

Reviewer 2: ```suggestion
// returns environment PORT or port parameter, both will be prefixed with ":"
```
